internal/resourcemodifiers/resource_modifiers.go

```diff
@@ -117,7 +118,24 @@ func (r *ResourceModifierRule) PatchArrayToByteArray() ([]byte, error) {
 }
 
 func (p *JSONPatch) ToString() string {
-	if strings.Contains(p.Value, "\"") {
+	// if value is empty, then add empty quotes
+	if p.Value == "" {
+		return fmt.Sprintf(`{"op": "%s", "from": "%s", "path": "%s", "value": ""}`, p.Operation, p.From, p.Path)
+	}
+	// if value is null, then don't add quotes
+	if p.Value == "null" {
+		return fmt.Sprintf(`{"op": "%s", "from": "%s", "path": "%s", "value": %s}`, p.Operation, p.From, p.Path, p.Value)
+	}
+	// if value is a boolean, then don't add quotes
+	if _, err := strconv.ParseBool(p.Value); err == nil {
+		return fmt.Sprintf(`{"op": "%s", "from": "%s", "path": "%s", "value": %s}`, p.Operation, p.From, p.Path, p.Value)
+	}
+	// if value is a json object or array, then don't add quotes
+	if strings.Contains(p.Value, "\"") || strings.Contains(p.Value, "[") || strings.Contains(p.Value, "{") {
```

Comment:
here could we trigger false positives? like `"just\"string\""`, `today is Monday [12/09]`

Reply:
we could, but I am relying on assumption that most YAMLs don't generally keep values such as this.
On the other hand didn't want to overengineer this to allow false positives to go in as is.
Any suggestions on how we could fix it. Meanwhile I'll also think further and see if this can be resolved.

Reply:
There could be other cases as well for boolean / null where the expecting k8s resource expects them inside quotes instead of bool / null

Reply:
Perhaps I can make this check as "startswith" "["  or "{"

Reply:
Agree, for now lets keep simple (loved the idea with `startswith`)